dotnet/samples/GettingStartedWithProcesses/Step02_AccountOpening.cs

```diff
@@ -0,0 +1,647 @@
+
+        // When the coreSystemRecordCreation step successfully creates a new accountId, it will trigger the creation of a new marketing entry through the marketingRecordCreation step
+        coreSystemRecordCreationStep
+            .OnEvent(AccountOpeningEvents.NewMarketingRecordInfoReady)
+            .SendEventTo(new ProcessFunctionTargetBuilder(marketingRecordCreationStep, functionName: "CreateNewMarketingEntry", parameterName: "userDetails"));
+
+        // When the coreSystemRecordCreation step successfully creates a new accountId, it will trigger the creation of a new CRM entry through the crmRecord step
+        coreSystemRecordCreationStep
+            .OnEvent(AccountOpeningEvents.CRMRecordInfoReady)
+            .SendEventTo(new ProcessFunctionTargetBuilder(crmRecordStep, functionName: "CreateCRMEntry", parameterName: "userInteractionDetails"));
+
+        // When the coreSystemRecordCreation step successfully creates a new accountId, it will pass the account information details to the welcomePacket step
+        coreSystemRecordCreationStep
+            .OnEvent(AccountOpeningEvents.NewAccountDetailsReady)
+            .SendEventTo(new ProcessFunctionTargetBuilder(welcomePacketStep, parameterName: "accountDetails"));
+
+        // When the marketingRecordCreation step successfully creates a new marketing entry, it will notify the welcomePacket step it is ready
+        marketingRecordCreationStep
+            .OnEvent(AccountOpeningEvents.NewMarketingEntryCreated)
+            .SendEventTo(new ProcessFunctionTargetBuilder(welcomePacketStep, parameterName: "marketingEntryCreated"));
+
+        // When the crmRecord step successfully creates a new CRM entry, it will notify the welcomePacket step it is ready
+        crmRecordStep
+            .OnEvent(AccountOpeningEvents.CRMRecordInfoEntryCreated)
+            .SendEventTo(new ProcessFunctionTargetBuilder(welcomePacketStep, parameterName: "crmRecordCreated"));
+
+        // After crmRecord and marketing gets created, a welcome packet is created to then send information to the user with the mailService step
+        welcomePacketStep
+            .OnEvent(AccountOpeningEvents.WelcomePacketCreated)
+            .SendEventTo(new ProcessFunctionTargetBuilder(mailServiceStep, functionName: "SendMailToUserWithDetails", parameterName: "message"));
+
+        // All possible paths end up with the user being notified about the account creation decision throw the mailServiceStep completion
+        mailServiceStep
+            .OnEvent(AccountOpeningEvents.MailServiceSent)
+            .StopProcess();
+
+        KernelProcess kernelProcess = process.Build();
+
+        var runningProcess = await LocalKernelProcessFactory.StartAsync(kernelProcess, kernel, new KernelProcessEvent() { Id = "StartProcess", Data = null });
+    }
+
+    public class UserInputStep : KernelProcessStep<UserInputState>
```

Comment:
Can you make all these classes private and sealed?

Reply:
done in the remaining steps in the Step02 file and the rest are in separate files now